File: Dia13/CalculadoraPropina.py
# Calculadora de Propina en Python

# "values" es una lista
def tipCalculator(values, percentage):
    sumValues = 0
    for i in range(0, len(values)):
        sumValues += values[i]
    return sumValues * percentage

valores = []
i = 0
otroPlatillo = True

while(otroPlatillo):
    i += 1
    valor = float(input('Platillo {}: ¿Cuánto valió? '.format(i)))
    valores.append(valor)
    otroPlatillo = (input('¿Hay más platillos? (si/no): ') == 'si')

porcentaje = float(input('¿Cuál será el procentaje de la propina? (1 - 100%)? ')) / 100
print('La propina es de {}'.format(tipCalculator(valores, porcentaje)))

File: Dia13/CalculadoraPropina.js
//Calculadora de Propina en JavaScript

//"values" es un Array
function tipCalculator(values, percentage) {
    let sumValues = values.reduce((a, c) => a + c)
    return sumValues * percentage
}

let valores = []
let i = 0
do {
    i++
    let valorPlatillo = parseFloat(prompt(`Platillo ${i}: ¿Cuánto valió?`))
    valores.push(valorPlatillo)
}while(prompt('¿Hay más platillos? (si/no)').toLowerCase() == 'si')

let porcentajePropina = parseFloat(prompt('¿Cuál será el procentaje de la propina? (1 - 100%)')) / 100
alert(`La propina es de ${tipCalculator(valores, porcentajePropina)}`)